test/rateLimiters/slidingWindowLog.test.ts

```diff
@@ -0,0 +1,568 @@
+                timestamp: timestamp + 100,
+                tokens: requested,
+            },
+        ]);
+        expect(await getLogFromClient(client, user2)).toEqual([
+            {
+                timestamp: timestamp + 200,
+                tokens: 1,
+            },
+        ]);
+        expect(await getLogFromClient(client, user3)).toEqual([{ timestamp, tokens: requested }]);
+    });
+
+    test('is able to be reset', async () => {
+        const tokens = 5;
+        await setLogInClient(client, user1, [{ tokens, timestamp }]);
+        await setLogInClient(client, user2, [{ tokens, timestamp }]);
+        await setLogInClient(client, user3, [{ tokens, timestamp }]);
+
+        limiter.reset();
+
+        expect(getLogFromClient(client, user1)).resolves.toEqual([]);
+        expect(getLogFromClient(client, user2)).resolves.toEqual([]);
+        expect(getLogFromClient(client, user3)).resolves.toEqual([]);
+
+        expect((await limiter.processRequest(user1, timestamp, CAPACITY)).success).toBe(true);
+        expect((await limiter.processRequest(user2, timestamp, CAPACITY - 1)).success).toBe(true);
+        expect((await limiter.processRequest(user3, timestamp, CAPACITY + 1)).success).toBe(false);
+    });
+
+    describe('is configurable...', () => {
+        test('does not allow capacity or window size <= 0', () => {
+            expect(() => new SlidingWindowLog(0, 1, client)).toThrow(
+                'SlidingWindowLog window size and capacity must be positive'
+            );
+            expect(() => new SlidingWindowLog(-10, 1, client)).toThrow(
+                'SlidingWindowLog window size and capacity must be positive'
+            );
+            expect(() => new SlidingWindowLog(10, -1, client)).toThrow(
+                'SlidingWindowLog window size and capacity must be positive'
+            );
+            expect(() => new SlidingWindowLog(10, 0, client)).toThrow(
+                'SlidingWindowLog window size and capacity must be positive'
+            );
+        });
+
+        xtest('...allows custom window size and capacity', async () => {
+            const customWindow = 500;
+            const customSizelimiter = new SlidingWindowLog(customWindow, CAPACITY, client);
+
+            let customSizeSuccess = await customSizelimiter
+                .processRequest(user1, timestamp, CAPACITY)
+                .then((res) => res.success);
+            expect(customSizeSuccess).toBe(true);
+
+            customSizeSuccess = await customSizelimiter
+                .processRequest(user1, timestamp + 100, CAPACITY)
+                .then((res) => res.success);
+            expect(customSizeSuccess).toBe(false);
+
+            customSizeSuccess = await customSizelimiter
+                .processRequest(user1, timestamp + customWindow, CAPACITY)
+                .then((res) => res.success);
+
+            const customCapacitylimiter = new SlidingWindowLog(WINDOW_SIZE, 5, client);
+            const customCapacity = 5;
```

Comment:
super nitpicky, but you could switch line 555 with 554 and use the `customCapacity` constant to instantiate the ratelimiter.